### PR TITLE
Working scan

### PR DESCRIPTION
**Pull request by namebee**

### CODE REVIEW

File: src/main/java/io/github/hydos/proton/Proton.java [hunk 1]
```diff
@@ -3,11 +3,12 @@
 import io.github.hydos.proton.module.ModuleManager;
 import io.github.hydos.proton.sound.Sound;
 import net.fabricmc.api.ModInitializer;
+import net.fabricmc.loader.api.entrypoint.PreLaunchEntrypoint;
 import org.apache.logging.log4j.Level;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-public class Proton implements ModInitializer {
+public class Proton implements ModInitializer, PreLaunchEntrypoint {
```
Comment (YTG1234) [inline src/main/java/io/github/hydos/proton/Proton.java]: I prefer having a separate class for that

File: src/main/java/io/github/hydos/proton/module/Module.java [hunk 2]
```diff
@@ -1,20 +1,17 @@
...
-import net.minecraft.util.Identifier;
-
 public abstract class Module {
-
-    public final Identifier id;
     public final String name;
+    public final String id;
 
-    public Module(String name, Identifier id){
+    public Module(String name, String id){
```
Comment (YTG1234) [inline src/main/java/io/github/hydos/proton/module/Module.java]: why :concern:

File: src/main/java/io/github/hydos/proton/module/ModuleManager.java [hunk 3]
```diff
@@ -27,6 +35,21 @@ public void setupCommonModules() {
...
 
+    public void scanAndRegisterModules() {
+        ScanResult scanner = new FastClasspathScanner().scan();
+        List<String> classes = scanner.getNamesOfSubclassesOf(Module.class);
+
+        for (String classname : classes) {
+            try {
+                Class<?> clazz = Class.forName(classname);
+                Module module = (Module) clazz.getDeclaredConstructor().newInstance();
+                this.addModule(module);
```
Comment (YTG1234) [inline src/main/java/io/github/hydos/proton/module/ModuleManager.java:46]: How would that work with a config?

**Review by YTG1234 — CHANGES_REQUESTED** [review]

**Comment by mounderfod** [comment]
Why do we even need scanning? The current system works fine.

**Comment by YTG1234** [comment]
Overall I like the idea of this PR, but for the reasons above I will not be approving it

**Comment by YTG1234** [comment]
> Why do we even need scanning? The current system works fine.

It's cool, and we don't have to manually register each module

**Review by YTG1234 — COMMENTED** [review]
👍

**Comment by YTG1234** [comment]
I resolved the conflicts, but I still have the config problem

**Pull request merged**